1.4/Source/Bullet_Impact_Spread.cs

```diff
@@ -0,0 +1,59 @@
+﻿using HarmonyLib;
+using RimWorld;
+using Verse;
+using UnityEngine;
+using System.Collections.Generic;
+using System;
+using VFECore;
+
+namespace VPE_Ranger
+{
+    [HarmonyPatch(typeof(Bullet))]
+    [HarmonyPatch("Impact")]
+    public static class Bullet_Impact_Spread
+    {
+        private static void Postfix(ref Thing hitThing, Bullet __instance)
+        {
+            if (__instance == null || __instance.Launcher == null || !(__instance.Launcher is Pawn pawn))
+            {
+                return;
+            }
+            //if (pawn.equipment.Primary.def.IsMeleeWeapon) return;
+
+            if (pawn.health.hediffSet.HasHediff(VPERanger_DefOf.MakaiRanger_BuffThree))
+            {
+                HediffComp_ArcingShot comp = pawn.health.hediffSet.GetFirstHediffOfDef(VPERanger_DefOf.MakaiRanger_BuffThree).TryGetComp<HediffComp_ArcingShot>();
+                if (comp.shotCount > 0 && !comp.isBursted)
+                {
+                    List<IntVec3> possibleTargetCell = new List<IntVec3>();
+                    int count = Mathf.RoundToInt(5 * pawn.GetStatValue(StatDefOf.PsychicSensitivity));
+                    for (int i = 0; i < count; i++)
+                    {
+                        possibleTargetCell.Add(VPERangerUtility.RandomCellAroundCellBase(__instance.intendedTarget.Cell, -5, 5));
+                    }
+                    for (int i = 0; i < possibleTargetCell.Count; i++)
+                    {
+                        Projectile shrapnelThing = (Projectile)GenSpawn.Spawn(__instance.def, __instance.Position, __instance.Launcher.Map);
+                        Thing possibleThing = possibleTargetCell[i].GetFirstPawn(__instance.Launcher.Map);
+                        Thing possibleBuilding = possibleTargetCell[i].GetFirstBuilding(__instance.Launcher.Map);
+                        if (possibleThing != null)
+                        {
+                            shrapnelThing.Launch(__instance.Launcher, possibleThing, possibleThing, ProjectileHitFlags.IntendedTarget);
+                        }
+                        else if (possibleBuilding != null)
+                        {
+                            shrapnelThing.Launch(__instance.Launcher, possibleBuilding, possibleBuilding, ProjectileHitFlags.IntendedTarget);
+                        }

```

Comment:
No need to get the building if you find a thing on the first try so you can use `??` to grab the building only if needed.

```suggestion
                        Thing possibleThing = possibleTargetCell[i].GetFirstPawn(__instance.Launcher.Map) ?? possibleTargetCell[i].GetFirstBuilding(__instance.Launcher.Map);
                        if (possibleThing != null)
                        {
                            shrapnelThing.Launch(__instance.Launcher, possibleThing, possibleThing, ProjectileHitFlags.IntendedTarget);
                        }
```